control/statefbk.py

```diff
@@ -347,6 +347,9 @@ def gram(sys,type):
     #Check for ss system object
     if not isinstance(sys,statesp.StateSpace):
         raise ValueError("System must be StateSpace!")
+    gramtypes = ['c', 'o', 'cf', 'of']
```

Comment:
Why do you declare `gramtypes` as a variable? It only appears to be used in the conditional expression on the next line, so the motivation for allocating a variable with function-level scope is not apparent to me.